dashboard/app/controllers/api_controller.rb

```diff
@@ -153,6 +153,105 @@ def section_text_responses
     render json: data
   end
 
+  def section_assessments
+    load_section
+    load_script
+
+    level_group_script_levels = @script.script_levels.includes(:level).where('levels.type' => LevelGroup)
+
+    data = @section.students.map do |student|
+      student_hash = {id: student.id, name: student.name}
+
+      level_group_script_levels.map do |script_level|
+        next unless script_level.long_assessment?
+
+        last_attempt = student.last_attempt(script_level.level)
+        response = last_attempt.try(:level_source).try(:data)
+
+        next unless response
+
+        user_level = UserLevel.where(user_id: student.id, level: script_level.level, script: @script).first
+        puts user_level.to_json
+
+        # Summarize some key data.
+        multi_count = 0
+        multi_count_correct = 0
+
+        # And construct a listing of all the individual levels and their results.
+        level_results = []
+
+        next unless properties = script_level.level[:properties]
+
+        # Go through each page of the long assessment's LevelGroup.
+        properties["pages"].each do |page|
+          # Go through each level of the page.
+          page["levels"].each do |page_level_name|
+            level = Level.find_by_name(page_level_name)
+
+            # And work with the response to the individual level inside the LevelGroup.
+            response_parsed = JSON.parse(response)
```

Comment:
Maybe pull this parse call up a couple levels (maybe close to where you get response out of the last_attempt) so it does not get called n*m times?